hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/SelectClause.java

```diff
@@ -208,9 +208,7 @@ public void initializeExplicitSelectClause(FromClause fromClause) throws Semanti
 					else {
 						origin = fromElement.getRealOrigin();
 					}
-					if ( !fromElementsForLoad.contains( origin )
-							// work around that fetch joins of element collections where their parent instead of the root is selected
-							&& ( !fromElement.isCollectionJoin() || !fromElementsForLoad.contains( fromElement.getFetchOrigin() ) ) ) {
+					if ( !fromElementsForLoad.contains( origin ) && !fromElementsForLoad.contains( fromElement.getFetchOrigin() ) ) {
```

Comment:
could we create some testing case for this? Otherwise it seems not that compelling.

Reply:
saw the attachment in the ticket. Would you mind my repeating the process again to create a new PR including the testing case?

Reply:
That would be great :) 👍

Reply:
I created a new PR at https://github.com/hibernate/hibernate-orm/pull/3482